pkg/collector/py/datadog_agent.c

```diff
@@ -181,8 +180,8 @@ static PyObject *add_external_tags(PyObject *self, PyObject *args) {
         }
 
         // copy the list of tags into an array of char*
-        int actual_size = 0;
-        for (int j=0; j<tags_len; j++) {
+        int j, actual_size = 0;
+        for (j=0; j<tags_len; j++) {
             PyObject *s = PyList_GetItem(value, j);
```

Comment:
I just re-read this, and I think the logic is incorrect (and will definitely cause crashes in the cleanup) in the case where `PyList_GetItem()` or `PyStringAsString()` returns `NULL`.  Consider tags_len of 5, but the 3rd entry comes back `null` for some reason.  The `tags` array will be
`tag[0] = <something>`
`tag[1] = <something>`
`tag[2] = <unintialized, best case NULL>`
`tag[3] = <something>`
`tag[4] = <something>`
And `actual_size` will be 4.
This will cause `AddExternalTags` to get an array with length 4 with a hole in the middle.
Then, in `//cleanup` below, you'll call `free()` on `NULL` (or uninitialized memory), and leak `tag[4]`